### PR TITLE
DAT-2899 add context do all media

### PR DESCRIPTION
**Pull request by dianafa**
https://wikia-inc.atlassian.net/browse/DAT-2899

ACHTUNG! Do not merge without the mercury part: https://github.com/Wikia/mercury/pull/987

edit: tests added, one more LGTM needed :) ping @RafLeszczynski / @idradm / @SebastianMarzjan

### CODE REVIEW

File: extensions/wikia/ArticleAsJson/ArticleAsJson.class.php [hunk 1]
```diff
@@ -7,8 +7,15 @@ class ArticleAsJson extends WikiaService {
...
 	];
 
+	const ICON_MAX_SIZE = 40;
 	const CACHE_VERSION = '0.0.3';
 
+	const ARTICLE_IMAGE = 'article-image';
+	const ARTICLE_VIDEO = 'article-video';
+	const GALLERY_IMAGE = 'gallery-image';
+	const ICON = 'icon';
+	const INFOBOX_IMAGE = 'infobox-image';
```
Comment (SebastianMarzjan) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: I'm thinking about making the contexts prefixed by IMAGE_CONTEXT_ or something along similar lines. It'll be especially confusing when seeing ICON being used for the first time/

File: extensions/wikia/ArticleAsJson/ArticleAsJson.class.php [hunk 2]
```diff
@@ -156,6 +165,11 @@ public static function onImageBeforeProduceHTML( &$dummy, Title &$title, &$file,
 
 			$details = WikiaFileHelper::getMediaDetail( $title, self::$mediaDetailConfig );
 
+			//information for mobile skins how they should display small icons
+			$fixedWidth = isset($handlerParams['width']) ? $handlerParams['width'] < self::ICON_MAX_SIZE : false;
```
Comment (SebastianMarzjan) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: Can we extract this logic to something leading in the direction of:
```
$details['context'] = self::isFixedImage($handlerParams) ? self::ICON : self::ARTICLE_IMAGE
```

File: extensions/wikia/ArticleAsJson/ArticleAsJson.class.php [hunk 3]
```diff
@@ -156,6 +165,11 @@ public static function onImageBeforeProduceHTML( &$dummy, Title &$title, &$file,
...
+			$fixedHeight = isset($handlerParams['height']) ? $handlerParams['height'] < self::ICON_MAX_SIZE : false;
+			$details['context'] = ($fixedWidth || $fixedHeight) ? self::ICON : ARTICLE_IMAGE;
```
Comment (SebastianMarzjan) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: shouldn't ARTICLE_IMAGE be self:: ARTICLE_IMAGE ?

File: extensions/wikia/ArticleAsJson/ArticleAsJson.class.php [hunk 4]
```diff
@@ -277,4 +289,16 @@ private static function linkifyMediaCaption( Parser $parser, &$media ) {
...
 		}
 	}
+
+	/**
+	 * @param $handlerParams
+	 * @desc Determines if height or width of image was set
+	 * explicitly by user by adding '{width}px' or 'x{height}px' to image wikitext
+	 * @return bool true if the image size was manipulated by user
+	*/
+	private static function isFixedImage( $handlerParams ) {
```
Comment (idradm) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: Could we make this method name more descriptive?

Comment (SebastianMarzjan) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: Yup, this method actually means something along the lines of isIcon or isFormattedAsIcon

**Comment by RafLeszczynski** [comment]
Shouldn't we update / create some unit / integration tests after these changes ? Except that LGTM :-)

File: extensions/wikia/ArticleAsJson/ArticleAsJson.class.php [hunk 5]
```diff
@@ -277,4 +289,16 @@ private static function linkifyMediaCaption( Parser $parser, &$media ) {
 			$parser->replaceLinkHolders( $media['caption'] );
 		}
 	}
+
+	/**
+	 * @param $handlerParams
+	 * @desc Determines if height or width of image was set
+	 * explicitly by user by adding '{width}px' or 'x{height}px' to image wikitext
```
Comment (rogatty) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: This description isn't true (there is also max size). It should also be above `@param`.

**Comment by RafLeszczynski** [comment]
LGTM

**Pull request merged**